libs/langchain/langchain/vectorstores/redis/filters.py

```diff
@@ -0,0 +1,429 @@
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("zipcode") == 90210
+        """
+        self._set_value(other, int, RedisFilterOperator.EQ)
+        return RedisFilterExpression(str(self))
+
+    @check_operator_misuse
+    def __ne__(self, other: int) -> "RedisFilterExpression":
+        """Create a Numeric inequality filter expression
+
+        Args:
+            other (int): The value to filter on.
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("zipcode") != 90210
+        """
+        self._set_value(other, int, RedisFilterOperator.NE)
+        return RedisFilterExpression(str(self))
+
+    def __gt__(self, other: int) -> "RedisFilterExpression":
+        """Create a RedisNumeric greater than filter expression
+
+        Args:
+            other (int): The value to filter on.
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("age") > 18
+        """
+        self._set_value(other, int, RedisFilterOperator.GT)
+        return RedisFilterExpression(str(self))
+
+    def __lt__(self, other: int) -> "RedisFilterExpression":
+        """Create a Numeric less than filter expression
+
+        Args:
+            other (int): The value to filter on.
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("age") < 18
+        """
+        self._set_value(other, int, RedisFilterOperator.LT)
+        return RedisFilterExpression(str(self))
+
+    def __ge__(self, other: int) -> "RedisFilterExpression":
+        """Create a Numeric greater than or equal to filter expression
+
+        Args:
+            other (int): The value to filter on.
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("age") >= 18
+        """
+        self._set_value(other, int, RedisFilterOperator.GE)
+        return RedisFilterExpression(str(self))
+
+    def __le__(self, other: int) -> "RedisFilterExpression":
+        """Create a Numeric less than or equal to filter expression
+
+        Args:
+            other (int): The value to filter on.
+
+        Example:
+            >>> from langchain.vectorstores.redis import RedisNum
+            >>> filter = RedisNum("age") <= 18
+        """
+        self._set_value(other, int, RedisFilterOperator.LE)
+        return RedisFilterExpression(str(self))
+
+
+class RedisText(RedisFilterField):
+    """A RedisText is a RedisFilterField representing a text field in a Redis index."""
+
+    OPERATORS = {
+        RedisFilterOperator.EQ: "==",
+        RedisFilterOperator.NE: "!=",
+        RedisFilterOperator.LIKE: "%",
+    }
+    OPERATOR_MAP = {
+        RedisFilterOperator.EQ: '@%s:"%s"',
+        RedisFilterOperator.NE: '(-@%s:"%s")',
+        RedisFilterOperator.LIKE: "@%s:%s",
```

Comment:
I'm not sure that LIKE represents what you meant it would be. The difference between `@%s:"%s"` and `@%s:%s` is that the first one allows you to find exact matches **for phrases**, while the last one also looks for exact matches but for each token separately. Also, using `@%s:%s` (without quotes) will support [stemming](https://redis.io/docs/interact/search-and-query/advanced-concepts/stemming/) by default.
To get a behavior that is more similar to SQL's LIKE operator, we can use prefix, infix or suffix matching (see [query](https://redis.io/docs/interact/search-and-query/query/) docs).